include/sys/abd.h

```diff
@@ -56,8 +60,13 @@ typedef struct abd {
 	union {
 		struct abd_scatter {
 			uint_t		abd_offset;
+#if defined(__FreeBSD__) && defined(_KERNEL)
+			uint_t  abd_chunk_size;
+			void    *abd_chunks[];
+#else
 			uint_t		abd_nents;
 			struct scatterlist *abd_sgl;
+#endif
 		} abd_scatter;
```

Comment:
I do wonder if `#if linux` around the `scatterlist` wouldn't be easier, or it'll be
`#if ( defined(__FreeBSD__) || defined(__APPLE__) || defined(_WIN32)) && defined(_KERNEL) `
But don't change anything, just curious what people think, is reducing #ifdefs even on the agenda just yet? I'm probably here waaay too early

Reply:
@lundman your timing is perfect!  Reducing the `#ifdefs` is definitely on the agenda as is any additional refactoring we should be considering to enable adding additional platforms.